great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -1019,6 +1105,23 @@ def run_profiler_on_data(
         )
         return result
 
+    def add_rule(self, rule: Rule) -> None:
+        """
+        Add Rule object to existing profiler object by appending to self._rules. In cases where a Rule with the same name exists, it is updated.
+        """
+        if not isinstance(rule, Rule):
```

Comment:
@Shinnnyshinshin If `rule` is already declared as `Rule` type, then this check should not be here.  If, however, the type were a `Union`, that would be a different case.  Thanks.

Reply:
Done :) Thank you